public/app/features/explore/QueryLibrary/QueryTemplatesList.tsx

```diff
@@ -1,14 +1,101 @@
+import { css } from '@emotion/css';
+import { uniqBy } from 'lodash';
+import { useEffect, useMemo, useState } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { EmptyState, FilterInput, InlineLabel, MultiSelect, Spinner, useStyles2, Stack } from '@grafana/ui';
+import { t, Trans } from 'app/core/internationalization';
+import { createQueryText } from 'app/core/utils/richHistory';
 import { useAllQueryTemplatesQuery } from 'app/features/query-library';
 import { QueryTemplate } from 'app/features/query-library/types';
 
 import { getDatasourceSrv } from '../../plugins/datasource_srv';
 
+import { QueryLibraryProps } from './QueryLibrary';
 import QueryTemplatesTable from './QueryTemplatesTable';
 import { QueryTemplateRow } from './QueryTemplatesTable/types';
+import { searchQueryLibrary } from './utils/search';
+
+interface QueryTemplatesListProps extends QueryLibraryProps {}
 
-export function QueryTemplatesList() {
+export function QueryTemplatesList(props: QueryTemplatesListProps) {
   const { data, isLoading, error } = useAllQueryTemplatesQuery();
+  const [searchQuery, setSearchQuery] = useState('');
+  const [datasourceFilters, setDatasourceFilters] = useState<Array<SelectableValue<string>>>(
+    props.activeDatasources?.map((ds) => ({ value: ds, label: ds })) || []
+  );
+
+  const [allQueryTemplateRows, setAllQueryTemplateRows] = useState<QueryTemplateRow[]>([]);
+  const [isRowsLoading, setIsRowsLoading] = useState(true);
+  const styles = useStyles2(getStyles);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    const fetchRows = async () => {
+      if (!data) {
+        setIsRowsLoading(false);
+        return;
+      }
+
+      try {
+        const rowsPromises = data.map(async (queryTemplate: QueryTemplate, index: number) => {
+          const datasourceRef = queryTemplate.targets[0]?.datasource;
+          const datasourceApi = await getDataSourceSrv().get(datasourceRef);
+          const datasourceType = getDatasourceSrv().getInstanceSettings(datasourceRef)?.meta.name || '';
+          const query = queryTemplate.targets[0];
+          const queryText = createQueryText(query, datasourceApi);
+          const datasourceName = datasourceApi?.name || '';
+
+          return {
+            index: index.toString(),
+            uid: queryTemplate.uid,
+            datasourceName,
+            datasourceRef,
+            datasourceType,
+            createdAtTimestamp: queryTemplate?.createdAtTimestamp || 0,
+            query,
+            queryText,
+            description: queryTemplate.title,
+            user: queryTemplate.user,
+          };
+        });
+
+        const rows = await Promise.all(rowsPromises);
+
+        if (isMounted) {
+          setAllQueryTemplateRows(rows);
+          setIsRowsLoading(false);
+        }
+      } catch (error) {
+        console.error('Error fetching query template rows:', error);
```

Comment:
First I think we should at least show an alert if there is an error. Other thing is maybe we can move it inside the row map so that if we could not fetch data for one dataSource we don't throw away all the rows.